integration/spark2/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala

```diff
@@ -110,8 +110,16 @@ object CarbonFilters {
     }
 
     def getCarbonLiteralExpression(name: String, value: Any): CarbonExpression = {
-      new CarbonLiteralExpression(value,
-        CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name)))
+      val dataTypeOfAttribute = CarbonScalaUtil.convertSparkToCarbonDataType(dataTypeOf(name))
+      val dataType = if (Option(value).isDefined
+        && dataTypeOfAttribute == DataType.STRING
+        && value.isInstanceOf[Double]) {
```

Comment:
In the scenarios where we have stored the numbers as String (and have included it as a Dimension) and if we are trying to do operations like >= or <= or between it is not working correctly 
and it gives some random results as it is comparing a String with a Double. So for those cases this fix will work.